### DISCUSSION TITLE
condition explanation

### DISCUSSION THREAD

**Discussion by elerik** [OP-TED/eForms-SDK#725] [Q&A] [answered]
Hi devs,

with version 1.9 i have noticed this kind of condition (which id didnt noticed in previous version may be so i think about it right now) 

BT-161-NoticeResult

condition 
not(BT-142-LotResult[BT-13713-LotResult in BT-137-Lot[BT-765-Lot not in ('fa-mix','fa-w-rc','fa-wo-rc')]] == 'selec-w') and not(BT-768-Contract == TRUE)

specificaly first part is pretty wild
not(BT-142-LotResult[BT-13713-LotResult in BT-137-Lot[BT-765-Lot not in ('fa-mix','fa-w-rc','fa-wo-rc')]] == 'selec-w')

How to cut it from part to part to get true or false ? 
BT-765-Lot not in ('fa-mix','fa-w-rc','fa-wo-rc') and BT-137-Lot in BT-13713-LotResult and BT-142-LotResult !='selec-w'
something like that ?

**Comment by rouschr** [comment]
Hello,

> How to cut it from part to part to get true or false ?

Parsing this yourself is indeed challenging and not recommended, the best way to deal with all EFX conditions would be to use EFX:
https://github.com/OP-TED/efx-toolkit-java

I recommend you start by looking at the sdk1 unit tests:
https://github.com/OP-TED/efx-toolkit-java/blob/develop/src/test/java/eu/europa/ted/efx/sdk1/EfxExpressionTranslatorV1Test.java

If you use EFX you will benefit from our updates and bug fixes.

We have some ideas to demo things like this in the [editor demo project](https://github.com/OP-TED/eforms-notice-editor) but we do not really have time to do it right now. We already demo the use of CVS except that currently the output is only SVRL xml.

There is some work to use the meta info found in the SVRL and transform it to JSON to send back to the front-end: 
https://github.com/OP-TED/eforms-notice-editor/tree/feature/TEDEFO-1776-svrl-to-json
You can have a look at this to get some ideas, search for "svrl" in the .java files.
It parses the SVRL XML returned by CVS and does some transformations to prepare something nicer for the UI to consume in JSON.

**Reply by elerik** [reply]
Hi, 
well i do not use java. 
I just want to understand how to evaluate this more complex rule. 
To my eyes it looks like i wrote in first message. Are that sub conditions inside first [...] joined by AND ? 

Btw your svrl is completely fine and i use it succesfully with prolinks in to specific fields. If you will turn it to json will you keep also original svrl xml version ? For example use preferences how to return this response. Like "responseFormat": "xrvrlxml" eventualy json and developers could decide which kind of response they want ?

**Reply by rouschr** [reply, marked as answer]
> well i do not use java

The easiest solution would be to create a minimal Java server using Spring Boot + Maven, and use OpenJDK to run it, add dependencies to EFX toolkit, have one API end-point accepting HTTP requests with EFX text and responding with true or false.
Maybe the editor demo could have an API call like that, which you could call on localhost, currently the editor demo is not yet using EFX toolkit, but it could be an idea for later.

Another possibility is to start from the ANTLR grammar provided in the SDK: https://github.com/OP-TED/eForms-SDK/tree/develop/efx-grammar, but that is more difficult than trying to reuse the EFX toolkit.

> If you will turn it to json will you keep also original svrl xml version

CVS will still only return SVRL. The SVRL to JSON is only for the editor demo only (to give you some ideas).
The editor demo back-end calls CVS and gets an SVRL in response, then reads the SVRL XML and turns into JSON for the front-end of the editor demo.

**Reply by elerik** [reply]
Ok, 
well whatever i will make something with it. 
To CVS response. Perfect.
